pkgcfg/github.go

```diff
@@ -0,0 +1,89 @@
+package pkgcfg
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var baseConfigURL = "https://raw.githubusercontent.com/jetpack-io/devbox/main/pkgcfg/package-configuration"
+
+const githubDirContentAPI = "https://api.github.com/repos/jetpack-io/devbox/contents/pkgcfg/package-configuration"
+
+func getConfig(pkg, rootDir string) (*config, error) {
+	confURL, err := getBestConfigPath(pkg)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	} else if confURL == "" {
```

Comment:
what scenarios would lead to an empty confURL?

Reply:
if there's no json for that package. I guess there's technically a third error case where there may be a json but we errored out (possibly because of a network failure)

Reply:
edit: error case already handled, so all 3 cases handled correctly.